dialer.go

```diff
@@ -56,6 +56,12 @@ const (
 )
 
 var (
+	// ErrDialerClosed is used when a caller invokes Dial after closing the
+	// Dialer.
+	ErrDialerClosed = errors.New(
+		"Dialer has been closed. Close should be " +
+			"called only when a database connection is no longer needed.",
```

Comment:
usually errors aren't punctuated or capitalized, as they are chained together as a single sentence. for example, from std:

```
$ git grep -i 'errclosed.*errors\.new'
compress/lzw/reader.go:var errClosed = errors.New("lzw: reader/writer is closed")
database/sql/fakedb_test.go:var errClosed = errors.New("fakedb: statement has been closed")
internal/oserror/errors.go: ErrClosed     = errors.New("file already closed")
io/pipe.go:var ErrClosedPipe = errors.New("io: read/write on closed pipe")
net/http/h2_bundle.go:  http2errClosedPipeWrite        = errors.New("write on closed buffer")
net/http/h2_bundle.go:  http2errClosedBody         = errors.New("body closed by handler")
net/http/h2_bundle.go:var http2errClosedResponseBody = errors.New("http2: response body closed")
net/http/httputil/persist.go:var errClosed = errors.New("i/o operation on closed connection")
net/http/roundtrip_js.go:var errClosed = errors.New("net/http: reader is closed")
```

How about `alloydb: dialer is closed`? Given how almost every other close-able resource cannot be used once closed (files, connections, clients, etc), I don't think you need to explain when or when not to call close in the error message.

Reply:
I should know better :sweat_smile: Thanks for the suggestion. Will get that fixed.